### PR TITLE
Redeployability

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: _posts/2018-01-19-r14y.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+No two deployments of Taskcluster will be exactly alike -- that would defeat the purpose.
+We must support a limited amount of flexibility: which services are enabled, what features of those services are enabled, and credentials for the various cloud services we use.
+
+In some cases the configuration for a service relies on values derived from another service that must already be started.
+For example, the Queue needs Taskcluster credentials genereated by calling `createTask` on the a running Auth service.
+
+## Upgrades
+
+Many of the new features we have added in Taskcluster have been deployed through a carefully-choreographed, manual process.
+For exmaple, to deploy parameterized roles support, which involved a change to the Auth sevice's backend support, I disabled writes to the backend, carefully copied the data to the new backend, then landed a patch to begin using the new backend with the old frontend, and so on.
```
Comment (jonasfj) [inline _posts/2018-01-19-r14y.md]: example

**Review by jonasfj — APPROVED** [review]

File: _posts/2018-01-19-r14y.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+# Redeployability
+
+Redeployability means that Taskcluster can be deployed easily, multiple times, similar to OpenStack or Hadoop.
+If, when we finish this effort, there exist several distinct "instances" of Taskcluster in the world, then we have been successful.
+We will start by building a "staging" deployment of the Firefox instance, then move on to deploy instances that see production usage, first for other projects in Mozilla, and later for projects outside of Mozilla.
+
+In deciding to pursue this approach, we considered three options:
+
+ * Taskcluster as a service (TCaaS) -- we run the single global Taskcluster instance, providing that service to other projects just like Github or Travis-CI.
+ * Firefox CI -- Taskcluster persists only as a one-off tool to support Firefox's continuous integration system
```
Comment (petemoore) [inline _posts/2018-01-19-r14y.md:34]: persists -> exists ?

**Comment by petemoore** [comment]
The PR title should probably be R13y or Redeployability (I prefer the latter). :-)

File: _posts/2018-01-19-r14y.md (hunk 2 above)
Comment (djmitche) [reply _posts/2018-01-19-r14y.md:34]: I like "persists" because it suggests that the team dwindles to 1-2 people but doesn't go away.

**Comment by djmitche** [comment]
Sorry, I missed the PR title.  It's pretty clear that the team universally hates "r14y" now, I get it!  Not least because it's the wrong number :)

**Pull request merged**